src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -382,12 +392,20 @@ public int[] getRequiredTokens() {
     @Override
     public void visitToken(DetailAST ast) {
         final Details details = Details.getDetails(ast);
-        final DetailAST rcurly = details.rcurly;
 
-        if (rcurly != null) {
-            final String violation = validate(details);
-            if (!violation.isEmpty()) {
-                log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
+        if (details != null) {
+            final DetailAST rcurly = details.rcurly;
+            if (rcurly != null) {
+                final String violation = validate(details);
+                if (!violation.isEmpty()) {
+                    if (MSG_KEY_LINE_BREAK_AFTER.equals(violation)) {
+                        final DetailAST semiColon = Details.getNextToken(rcurly);
+                        log(semiColon, violation, ";", semiColon.getColumnNo() + 1);
+                    }
+                    else {
+                        log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
+                    }
+                }
```

Comment:
> @shubh220922 after looking at existing check logic (in main branch), it looks like we do the following for each token:
> 
> 1. Gather details
> 2. Check if rcurly exists (if not, no need to check anything since this is RightCurlyCheck)
> 3. Validate details based on information only in details
> 4. Place violations where appropriate
> 
> Now it looks like we are changing this pattern by adding a lot of logic in `visitToken` that is not immediately clear to the reader.
> 
> Are we able to keep existing pattern by determining everything we need from `Details`, and extending `validate` method only?

If we see, in the master branch, we only logged violation message for improper placement of ``}``(rightcurly token). However, in the first ever unique case of this PR, we have to log another violation message as well, which is releated to improper placement of ``;``(semicolon token).

Now the whole logic of the ``visitToken()`` has been centered trying to print the correct vioaltion message.

Reply:
After looking back at https://github.com/checkstyle/checkstyle/pull/12685#pullrequestreview-1430861266, I do not understand why we are returning `null` for curly braces; they exist for OBJBLOCK. Why don't we do something like:

1. Return details with associated rcurly
2. Have logic like:

```
        if (isAnonymousInnerClassRCurly(rcurly))) {
            // do stuff
            // log violations
        }
        else if (rcurly != null) {
            final String violation = validate(details);
            if (!violation.isEmpty()) {
                log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
            }
        }


```
?

Reply:
ok fix is in progress.